app/src/main/java/com/paranoidandroid/journey/wizard/utils/JourneyBuilder.java

```diff
@@ -85,6 +85,12 @@ public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
                                 break;
                             }
                         }
+
+                        // use first part of display name if no city is present
+                        if (destination.getCityName() == null) {
+                            String[] components = destination.getDisplayName().replaceAll("\\s+","").split(",");
```

Comment:
Note that this code is executing asynchronously. Are you relying on the fact that this runs sometime after onItemClick() returns?

Since onItemClick() and this callback (onSuccess()) must run on the event thread, then this code will run after the current call stack unwinds.

This method should work, but feels very dangerous. We can refactor in a separate PR if you'd like.